sockjsclient/xhr.go

```diff
@@ -172,23 +175,24 @@ func (x *XHRSession) handleResp(resp *http.Response) (msg string, again bool, er
 	case 'c':
 		x.mu.Lock()
 		x.opened = false
+		x.closed = true
 		x.mu.Unlock()
 
-		return "", false, errors.New("session closed")
+		return "", false, ErrSessionClosed
 	default:
 		return "", false, errors.New("invalid frame type")
 	}
 }
 
 func (x *XHRSession) Send(frame string) error {
-	if !x.isOpened() {
-		return errors.New("session is not opened yet")
-	}
-
 	if x.isClosed() {
 		return ErrSessionClosed
 	}
 
+	if !x.isOpened() {
```

Comment:
not related with this PR:

grammatically it should be `isOpen()` since it is adjective